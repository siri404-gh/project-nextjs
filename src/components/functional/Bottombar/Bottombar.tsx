import React, { FunctionComponent } from 'react';
import {
  BottomNavigation,
  BottomNavigationAction,
} from '@material-ui/core';

export type bottomListType = {
  title: string,
  img: string,
  route: string,
  links?: Array<{
    title: string,
    route: string,
  }>,
};

export interface Props {
  className?: string,
  list?: bottomListType[],
  value?: number,
  onChange?: (val) => void,
}

const Bottombar: FunctionComponent<Props> = ({ className, list, value, onChange, ...rest }) => {
  const _onChange = (event, val) => onChange(val);
  const getIcon = (icon, alt) => <img style={{ height: 25, width: 25, borderRadius: '50%' }} alt={alt} src={icon} />;
  const listKeyArr = Object.values(list);

  return listKeyArr.length > 0 && (
    <BottomNavigation
      {...rest}
      value={value}
      onChange={_onChange}
      className={className}>
      {Object.values(list).map((item: bottomListType) => (
        <BottomNavigationAction
          key={item.title}
          label={item.title}
          icon={getIcon(item.img, item.title)}
          style={{ minWidth: 25 }} />
      ))}
    </BottomNavigation>
  );
};

Bottombar.defaultProps = {
  className: '',
  list: [],
  value: 0,
  onChange: /* istanbul ignore next */ () => null,
};

export default Bottombar;
