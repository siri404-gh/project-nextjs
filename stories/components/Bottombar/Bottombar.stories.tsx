import React from 'react';
import { Story } from '@storybook/react/types-6-0';

import Bottombar, { Props } from '@/components/functional/Bottombar/Bottombar';

export default {
  title: 'Bottombar',
  component: Bottombar,
};

const Template: Story<Props> = args => <Bottombar {...args} />;

export const FirstStory = Template.bind({});

FirstStory.args = {
  list: [{
    title: '1',
    route: '/route',
    img: 'https://fakeimg.pl/300/',
  },
  {
    title: '2',
    route: '/route',
    img: 'https://fakeimg.pl/300/',
  }, {
    title: '3',
    route: '/route',
    img: 'https://fakeimg.pl/300/',
  }, {
    title: '4',
    route: '/route',
    img: 'https://fakeimg.pl/300/',
  }],
  value: 1,
  onChange: val => alert(val)
};
