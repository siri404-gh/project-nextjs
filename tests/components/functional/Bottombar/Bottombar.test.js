import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

import Bottombar from '@/components/functional/Bottombar/Bottombar';

describe('Bottombar', () => {

  let container;

  let props = {
    value: 0,
    className: 'class',
    onChange: jest.fn(),
    list: {
      'a': {
        img: 'content',
        title: 'title',
        route: '/route',
      },
    },
  };

  beforeEach(() => {
    container = document.createElement("section");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('renders without props', () => {
    act(() => {
      render(<Bottombar />, container);
    });

    expect(container).toMatchSnapshot();
  });

  it('renders with props', () => {
    act(() => {
      render(<Bottombar {...props} />, container);
    });

    expect(container).toMatchSnapshot();
  });

  it('should call onChange when button is clicked', () => {
    act(() => {
      render(<Bottombar {...props} />, container);
      const button = Array.from(document.querySelectorAll("button"))[0];
      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(props.onChange).toHaveBeenCalled();
  });

});
