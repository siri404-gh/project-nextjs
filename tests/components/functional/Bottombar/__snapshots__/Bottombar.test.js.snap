// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottombar renders with props 1`] = `
<section>
  <div
    class="MuiBottomNavigation-root class"
  >
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
      style="min-width: 25px;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <img
          alt="title"
          src="content"
          style="height: 25px; width: 25px; border-radius: 50%;"
        />
        <span
          class="MuiBottomNavigationAction-label Mui-selected"
        >
          title
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</section>
`;

exports[`Bottombar renders without props 1`] = `<section />`;
